### DISCUSSION TITLE
[Question] Understanding Tinygrad

### DISCUSSION THREAD

**Discussion by rozeappletree** [tinygrad/tinygrad#1543] [General]
Hi,

I have a couple of months of free time and wanted to do something worthwhile. What is the best way to understand tinygrad in and out? Quickly went through https://github.com/tinygrad/tinygrad/tree/master/docs and the abstractions file was interesting to read.

Additionally, I see a collection of 25 videos on youtube [(link here)](https://www.youtube.com/playlist?list=PLrUfsC4cOdl3MV_S_TWK_yAgcfVzMW805) ranging from 2y-5mo old videos, are they still relevant? 

Can I do something else apart from going through docs or YT videos to understand tinygrad better?

**Comment by salmanra** [comment]
I'm in a similar boat as you, read abstractions.py, wanted to learn more. [This](https://www.youtube.com/watch?v=Sk35MKtCXfQ&t=2640s&ab_channel=georgehotzarchive) tutorial video from three months ago walks through  beginner's details for running tinygrad tests and learning enough to contribute. 

Running tests, seeing what fails, and diagnosing why has proved a fast way to gain an understanding of tinygrad from the top down.

**Reply by salmanra** [reply]
Better yet, if you join the Discord server (there's a link in the [repo](https://github.com/tinygrad/tinygrad)), there are resources there to get learning.

**Comment by cornbutt** [comment]
@salmanra @INF800 any updates? hows it going? i want to learn enough to be able to contribute because george said once on a podcast that if you want a job at tiny corp you just need to contribute enough to the github. working on something even remotely impactful to the world is all i want with my career so i guess this is my shot lol.

**Reply by NinoRisteski** [reply]
any progress? want to do the same

**Reply by rozeappletree** [reply]
I think discord would be the best place to continue the discussion.

P.S. this is a very old post.
